### DISCUSSION TITLE
How to allow special characters in filenames on Linux?

### DISCUSSION THREAD

**Discussion by sojusnik** [aclap-dev/video-downloadhelper#1446] [Help and Q&A]
### Your question

Windows doesn't allow certain characters in its files, but Linux does. So is there a way to use such special characters when using this addon on Linux? Mostly it's about characters as `.`, `,`, `?`, `(`, `)` and the like, but also so-called umlauts, like ä, ö and ü, which are common f.i. in German.

### Addon & CoApp version and details

version: 9.1.0.44
target: mozilla
channel: stable
lang: de
coapp: {"found":true,"path":"/home/sojusnik/.local/share/vdhcoapp/vdhcoapp","version":"2.0.19","new_version":false}
license: {"unneeded":true}
platform: x86-64 linux
UA: Mozilla/5.0 (X11; Linux x86_64; rv:132.0) Gecko/20100101 Firefox/132.0

**Comment by GrampaWildWilly** [comment]
I do not work for VDH.  I'm just another user.

Of the characters you mentioned, only ? is not permitted in file names on Windows.  The others are all valid on Windows, including the umlauts.  The ? is not permitted because (I'm kind of guessing on this) it can be used as a wild card on the dir command, so it's essentially a reserved character.  If you want these special characters in your file names, you should switch from using the `Download` function in VDH to using the `Download As` function.  That way, you will have a chance to modify your file names to anything you like before the download proceeds.  I believe there are still characters that VDH will mishandle even if you do this.  But for the most part, it has worked fine for me.

There are some discussions in here about problems with Korean/Chinese/Japanese characters.  You can search for them if you're interested.  Our developer @paulrouget Paul said he would work on it.  But Paul hasn't posted here since early August.  I'm wondering if he still works for VDH.  That's not based on any inside information, of which I have none.  I'm just looking at his absence & voicing questions.

**Reply by sojusnik** [reply]
Thanks for this elaborate answer!

The proposal with `download as` would be only a temporary workaround, because you would have to rename each file manually that way, which is very time-consuming. Since I'm using `%title` in the `intelligent naming` menu (you can find it in the new version of CDH in the 3-dots-menu of an entry), I would prefer that VDH would apply special characters, like the ones I have mentioned, but others too, from the title of the webpage (or other variables,  when specified).
